### PR TITLE
MGDAPI-5142 backup/restore redis and postgres

### PR DESCRIPTION
**Pull request by cecobask**
# Issue link
https://issues.redhat.com/browse/MGDAPI-5142

# What
Investigation on how we can approach backup and restore of postgres and redis instances on gcp, using the gcloud cli.

# Verification steps
## Prerequisites
- Request GCP credentials
- Prepare the base cluster config using delorean
  ```
  make ocm/cluster.json BYOC=true CLOUD_PROVIDER=gcp OCM_CLUSTER_REGION=europe-west2
  ```
- Manually edit the cluster.json file to set the name, expiration timestamp and nodes to your preference
- Trigger cluster creation, wait for its completion and log in to it using the openshift cli
  ```
  make ocm/cluster/create
  ```
## Postgres
- Create an instance
  ```
  gcloud sql instances create test-postgres --region europe-west2 --database-version POSTGRES_14 --tier db-f1-micro
  ```
- Run the backup and restore script
  ```
  chmod +x ./test/scripts/backup-restore/gcp/postgres.sh
  INSTANCE_NAME=test-postgres ./test/scripts/backup-restore/gcp/postgres.sh
  ```
- The expected output should confirm that the backups are identical
- Delete the original postgres instance
  ```
  gcloud sql instances delete test-postgres --quiet
  ```
## Redis
- Create an instance
  ```
  gcloud redis instances create test-redis --region europe-west2
  ```
- Run the backup and restore script
  ```
  chmod +x ./test/scripts/backup-restore/gcp/redis.sh
  INSTANCE_NAME=test-redis ./test/scripts/backup-restore/gcp/redis.sh
  ```
- The expected output should confirm that the backups are identical
- Delete the original redis instance
  ```
  gcloud redis instances delete test-redis --region europe-west2 --quiet
  ```

### CODE REVIEW

**Comment by valerymo** [comment]
@cecobask , It's a very nice work; and it looks like a significant investigation work is hidden in these scripts.
Scripts are working and logs - attached.
I just wanted to add few minor comments; maybe some of them you will find useful.
1. Scripts names  - maybe use something like verify-postgres-backup-restore.sh, ... Although according to location it's clear that it's test and it's backup&restore, but seems to me it should be in name also.

2. Logs - seems to me, need show the Flow progress. When I read the script - it's clear for me, from comments, before each command. But when looking at screen , it's not so clear. So if possible to make Flow clear while looking at screen (without open the script) - it will be really nice. (+ final print on screen seems will be nice to have - test passed or not, to make it clear without looking at the whole flow : )

3. Script exit status? Do we need have exit status for automation, including the script in Pipeline, etc.

Just wanted to add again, that scripts are working fine and work is very good.
Thank you

[review_MGDAPI-5142_PostgresBackupRestoreTest.txt](https://github.com/integr8ly/integreatly-operator/files/10674409/review_MGDAPI-5142_PostgresBackupRestoreTest.txt)
[review_MGDAPI-5142_RedisBackupRestoreTest.txt](https://github.com/integr8ly/integreatly-operator/files/10674411/review_MGDAPI-5142_RedisBackupRestoreTest.txt)

File: test/scripts/backup-restore/gcp/postgres.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# compare the original and current state of backup files
+if cmp --silent -- "original.txt" "current.txt"; then
+  echo "postgres backups are identical"
+else
+  echo "postgres backups differ"
+fi
+# clean up resources
+rm original.sql original.txt current.sql current.txt
+gcloud storage rm --recursive gs://$BUCKET_NAME
+gcloud sql instances delete $RESTORED_INSTANCE_NAME --project $PROJECT_ID --quiet
```
Comment (valerymo) [inline test/scripts/backup-restore/gcp/postgres.sh:44]: work looks very nice for me, both scrips are working. Review test logs - attached at main page. Also few minor comments. Thank you

**Comment by cecobask** [comment]
Thank you for the review, @valerymo!
I've addressed your suggestions. Let me know if anything else can be changed 👍🏻

**Comment by valerymo** [comment]
@cecobask  thank you very much for addressing the comments
lgtm

**Pull request merged**